src/mame/drivers/mk98.cpp

```diff
@@ -0,0 +1,488 @@
+			m_scancode |= m_kdata;
+			// arrow keys
+			switch (m_scancode)
+			{
+				case 0x48: m_scancode = 0x3c; break;
+				case 0x50: m_scancode = 0x3e; break;
+				case 0x4b: m_scancode = 0x3d; break;
+				case 0x4d: m_scancode = 0x3f; break;
+				case 0x53: m_scancode = 0x3b; break;
+			}
+			LOGKBD("kbd: scancode %02x\n", m_scancode);
+			m_kbit = 0;
+			m_pic8259->ir1_w(ASSERT_LINE);
+		}
+		else
+		{
+			m_kbit++;
+		}
+	}
+
+	m_kclk = (state == ASSERT_LINE) ? true : false;
+}
+
+WRITE_LINE_MEMBER(mk98_state::keyboard_data_w)
+{
+	LOGKBD("kbd: KDATA: %d\n", state ? 1 : 0);
+	m_kdata = (state == ASSERT_LINE) ? 0x80 : 0x00;
+}
+
+/* video HLE */
+
+uint8_t mk98_state::video_register_r()
+{
+	uint8_t ret = 0;
+
+	switch (m_register_address_latch)
+	{
+	case 0x0e:
+		ret = (m_cursor_addr >> 8) & 0xff;
+		break;
+
+	case 0x0f:
+		ret = (m_cursor_addr >> 0) & 0xff;
+		break;
+
+	/* all other registers are write only and return 0 */
+	default:
+		break;
+	}
+
+	return ret;
+}
+
+void mk98_state::video_register_w(uint8_t data)
+{
+	switch (m_register_address_latch)
+	{
+	case 1:
+		m_graphics_mode = BIT(data, 4);
+		break;
+
+	case 0x0a:
+		m_cursor_start_ras = data & 0x7f;
+		break;
+
+	case 0x0c:
+		m_disp_start_addr = ((data & 0x3f) << 8) | (m_disp_start_addr & 0x00ff);
+		break;
+
+	case 0x0d:
+		m_disp_start_addr = ((data & 0xff) << 0) | (m_disp_start_addr & 0xff00);
+		break;
+
+	case 0x0e:
+		m_cursor_addr = ((data & 0x3f) << 8) | (m_cursor_addr & 0x00ff);
+		break;
+
+	case 0x0f:
+		m_cursor_addr = ((data & 0xff) << 0) | (m_cursor_addr & 0xff00);
+		break;
+	}
+}
+
+void mk98_state::video_address_w(uint8_t data)
+{
+	m_register_address_latch = data & 0x3f;
+}
+
+uint8_t mk98_state::video_r(offs_t offset)
+{
+	int data = 0xff;
+
+	switch (offset)
+	{
+	case 0:
+		return keyboard_r(0);
+
+	case 5:
+		data = video_register_r();
+		break;
+	}
+
+	return data;
+}
+
+void mk98_state::video_w(offs_t offset, uint8_t data)
+{
+	switch (offset)
+	{
+	case 2:
+		if (++m_font_upload > 63)
+			m_p_chargen[m_font_upload - 64] = data;
+		break;
+
+	case 4:
+		video_address_w(data);
+		break;
+
+	case 5:
+		video_register_w(data);
+		break;
+	}
+}
+
+
+void mk98_state::mk98_map(address_map &map)
+{
+	map.unmap_value_high();
+	map(0x00000, 0x1ffff).ram();
+	map(0xb8000, 0xbdfff).ram().share("video");
+	map(0xc0000, 0xdffff).noprw(); // ???
+	map(0xe0000, 0xfffff).rom().region("romdos", 0);
+}
+
+void mk98_state::mk98_io(address_map &map)
+{
+	map.unmap_value_low();
+	map(0x0000, 0x000f).unmaprw();
+	map(0x0020, 0x002f).rw(m_pic8259, FUNC(mk98pic_device::read), FUNC(mk98pic_device::write));
+	map(0x0040, 0x004f).rw("pit8254", FUNC(pit8254_device::read), FUNC(pit8254_device::write));
+	map(0x0060, 0x0063).rw(FUNC(mk98_state::keyboard_r), FUNC(mk98_state::keyboard_w));
+	map(0x00a0, 0x00a1).unmapw();
+	map(0x0110, 0x0111).unmapw();
+	map(0x0112, 0x0113).unmaprw();
+	map(0x0150, 0x0150).unmapw(); // cart slot select
+	map(0x0170, 0x0170).unmapw();
```

Comment:
What’s the point of these?  They aren’t actually punching holes in previously installed handlers.  If there’s something actually, present at that address, just use a comment.

Reply:
These are for documentation only, I'll convert them to comments